packages/peregrine/lib/Router/resolveUnknownRoute.js

```diff
@@ -99,6 +99,17 @@ function remotelyResolveRoute(opts) {
  * @returns {Promise<{type: "PRODUCT" | "CATEGORY" | "CMS_PAGE"}>}
  */
 function fetchRoute(opts) {
+    const storeCodes = AVAILABLE_STORE_VIEWS.map(store => store.code);
+    let route = opts.route;
+
+    storeCodes.forEach(function(storeCode) {
+        const path = '/' + storeCode.toLowerCase();
+        if (route.startsWith(path) || route === path) {
+            route = route.substring(path.length, route.length);
+        }
+    });
+    route = route === '' ? '/' : route;
```

Comment:
this is so magic and complex and relay readable

Reply:
I've updated this with a much simpler expression which results in the same functionality 😄